modules/util/sec_to_time.go

```diff
@@ -16,12 +16,24 @@ import (
 // 1563418		-> 2 weeks 4 days
 // 3937125s     -> 1 month 2 weeks
 // 45677465s	-> 1 year 6 months
+//
+// Magic numbers:
+// 3600 = 60 * 60 (amount of seconds in a hour)
+// 86400 = 60 * 60 * 24 (amount of seconds in a day)
 func SecToTime(duration int64) string {
 	formattedTime := ""
-	years := duration / (3600 * 24 * 7 * 4 * 12)
-	months := (duration / (3600 * 24 * 30)) % 12
-	weeks := (duration / (3600 * 24 * 7)) % 4
-	days := (duration / (3600 * 24)) % 7
+	years := (duration / 86400) / 365
+
+	// The following three variables are calculated with taking
+	// into account the previous calculated variables, this avoids
+	// pitfalls when using remainders. As that could lead to incorrect
+	// results when the calculated number equals the quotient number.
+	months := (duration/86400 - years*365) / 30
+	weeks := (duration/86400 - years*365 - months*30) / 7
+	days := duration/86400 - years*365 - months*30 - weeks*7
+
+	// The following three variables are calculated without depending
+	// on the previous calculated variables.
```

Comment:
These are more or less stable by definition, so to avoid multiplication by huge numbers. So leave it as-is.

Reply:
There is still an edge case, although it is not handled as badly as the ones that are being fixed here:

* 364 days are displayed as “12 months”
* 365 days are displayed as “1 year”

Not sure it is serious enough to worry about...